internal/pkg/gateway/api.go

```diff
@@ -427,47 +427,94 @@ func (gs *Server) Submit(ctx context.Context, request *gp.SubmitRequest) (*gp.Su
 		return nil, status.Errorf(codes.Unavailable, "no orderer nodes available")
 	}
 
-	// try each orderer in random order
 	var errDetails []proto.Message
-	for _, index := range rand.Perm(len(orderers)) {
-		orderer := orderers[index]
-		logger.Infow("Sending transaction to orderer", "txID", request.TransactionId, "endpoint", orderer.logAddress)
-
-		var response *ab.BroadcastResponse
-		var err error
-		done := make(chan struct{})
-		go func() {
-			defer close(done)
-			ctx, cancel := context.WithTimeout(ctx, gs.options.BroadcastTimeout)
-			defer cancel()
-
-			response, err = gs.broadcast(ctx, orderer, txn)
-		}()
-		select {
-		case <-done:
-			// Broadcast completed normally
-		case <-ctx.Done():
-			// Overall submit timeout expired
-			logger.Warn("Submit call timed out while broadcasting to ordering service")
-			return nil, newRpcError(codes.DeadlineExceeded, "submit timeout expired while broadcasting to ordering service")
+	if gs.getChannelConfig(request.ChannelId).ChannelConfig().Capabilities().ConsensusTypeBFT() {
```

Comment:
How is `gs.getChannelConfig(request.ChannelId)` implemented in practice? Are we going to the ledger? We shouldn't do I/O each time we send a transaction. 

If this requires I/O it makes sense to simply have a regular boolean field that will get loaded at initialization time for the time being.

Reply:
The config is cached in memory and only updated in response to config update